File: Exceções - IF Quest/Item.java
public class Item {
    private String nome="Sem nome";
    private int bonus;

    public Item(){

    }
    public Item(String n, int b){
        setNome(n);
        setBonus(b);
    }

    public String getNome(){
        return this.nome;
    }
    public int getBonus(){
        return this.bonus;
    }
    public void setNome(String n){
        this.nome = n;
    }
    public void setBonus(int b){
        if (b>=0){
            this.bonus = b;
        }
        else{
            throw new IllegalArgumentException("Bonus inválido, digite um valor maior que 0");
        }
    }

    public String descricao(){
        return getNome() + " (+" + getBonus() + ")";
    }
}

File: Exceções - IF Quest/item.py
class item():
    def __init__(self, n="Sem nome", b=0):
        self.nome=n
        self.bonus=b
    @property
    def nome(self):
        return self._nome
    @property
    def bonus(self):
        return self._bonus
    @nome.setter
    def nome(self, n=""):
        self._nome = n
    @bonus.setter
    def bonus(self, b):
        if (b>=0):
            self._bonus = b

    def descricao(self):
        return self._nome + " (+" + str(self._bonus) + ")"
